db/db_impl.cc

```diff
@@ -575,8 +575,8 @@ Status DBImpl::SetDBOptions(
                                             immutable_db_options_);
       env_options_for_compaction_.bytes_per_sync
                                     = mutable_db_options_.bytes_per_sync;
-      env_->OptimizeForCompactionTableWrite(env_options_for_compaction_,
-                                            immutable_db_options_);
+      env_options_for_compaction_ = env_->OptimizeForCompactionTableRead(
```

Comment:
should update env_options_for_compaction_.compaction_readahead_size here.

Reply:
Done